invenio_vocabularies/datastreams/readers.py

```diff
@@ -222,3 +225,56 @@ def _iter(self, fp, *args, **kwargs):
             raise ReaderError(f"Record not found in XML entry.")
 
         yield record
+
+
+class OAIPMHReader(BaseReader):
+    """OAIPMH reader."""
+
+    def __init__(
+        self,
+        *args,
+        base_url=None,
+        metadata_prefix=None,
+        set=None,
+        from_date=None,
+        until_date=None,
+        verb=None,
+        **kwargs,
+    ):
+        """Constructor."""
+        self._base_url = base_url
+        self._metadata_prefix = metadata_prefix if not None else "oai_dc"
+        self._set = set
+        self._until = until_date
+        self._from = from_date
+        self._verb = verb if not None else "ListRecords"
+        super().__init__(*args, **kwargs)
+
+    def _iter(self, scythe, *args, **kwargs):
+        """Read and parse an OAIPMH stream to dict."""
+        scythe.class_mapping["ListRecords"] = OAIRecord
+        try:
+            responses = scythe.list_records(
+                from_=self._from,
+                until=self._until,
+                metadata_prefix=self._metadata_prefix,
+                set_=self._set,
+                ignore_deleted=True,
+            )
+            for response in responses:
+                oaipmh_record = {
+                    "header": response.header,
+                    "metadata": response.metadata["record"],
```

Comment:
minor/question: is `response.header` an XML tree Python object, or a custom internal class of Scythe?

It might be an option to also just return `{"response": response}`, and leave it to the transformer (or the next reader) to extract whatever information is needed.

Reply:
Yes, `response.header` is an internal class (looked simply like a string at first look), as is `response`, too. If that's no problem you're right. I'll change that.